integr8ly/integreatly-operator
# expect new configmaps for alertmanager setup

currently there is a manual sop for setting up alertmanager config
via a secret

in this commit two new secrets are now expected by the operator:
- a secret for dead mans snitch url
- a secret for pagerduty service key

these secrets will be used by the operator to create a new config
secret for alertmanager called.

verification:
- run 'make code/run'
- ensure the 'redhat-rhmi-smtp' secret is created
- ensure the 'redhat-rhmi-pagerduty' secret is created
- ensure the 'redhat-rhmi-dead-mans-snitch' secret is created
- ensure the operator isn't halted waiting for the secrets
- ensure the monitoring stage completes successfully
- ensure the secret 'alertmanager-application-monitoring' exists
  in the monitoring operator namespace and follows the structure
  in https://issues.redhat.com/browse/INTLY-5200
- ensure the values from the three pre-requisite secrets are used
  in the new 'alertmanager-application-monitoring' secret data
- ensure the 'smtp to' email addresses in the config secret are
  formatted as noreply@<alertmanager-route>
- ensure the new config is visible through the ui of prometheus/
  alertmanager
- run 'ALERTING_EMAIL_ADDRESS=akeating@redhat.com make code/run'
- wait for the monitoring reconciliation to complete
- ensure the 'smtp to' email addresses in the config secret are
  'akeating@redhat.com'
